spec.bs

```diff
@@ -8159,43 +8313,155 @@ To <dfn>batch or fetch trusted bidding signals</dfn> given a [=trusted bidding s
+    1. Otherwise:
+      1. [=Fetch the current outstanding trusted signals batch=] given |trustedBiddingSignalsBatcher|,
+        |signalsUrl|, |scriptOrigin|, |experimentGroupId|, |topLevelOrigin|, |slotSizeQueryParam|, and
+        |policyContainer|.
+      1. If |ig|'s [=interest group/trusted bidding signals keys=] is not null, set
+        |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/keys=] to a
+        [=list/clone=] of |ig|'s [=interest group/trusted bidding signals keys=].
+      1. Otherwise, set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/keys=] to a
+        new [=list=].
       1. Set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/ig names=] to
-        |putativeIgNames|.
+          « |igName| ».
       1. Set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/length limit=] to
-        |putativeLengthLimit|.
+        |ig|'s [=interest group/max trusted bidding signals url length=].
   1. Otherwise:
-    1. [=Fetch the current outstanding trusted signals batch=] given |trustedBiddingSignalsBatcher|,
-      |signalsUrl|, |scriptOrigin|, |experimentGroupId|, |topLevelOrigin|, |slotSizeQueryParam|, and
-      |policyContainer|.
-    1. If |ig|'s [=interest group/trusted bidding signals keys=] is not null, set
-      |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/keys=] to a
-      [=list/clone=] of |ig|'s [=interest group/trusted bidding signals keys=].
-    1. Otherwise, set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/keys=] to a
-       new [=list=].
-    1. Set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/ig names=] to
-        « |igName| ».
-    1. Set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/length limit=] to
-      |ig|'s [=interest group/max trusted bidding signals url length=].
+    1. Let |keyValueInterestGroups| be |trustedBiddingSignalsBatcher|'s [=trusted bidding signals
+      batcher/key value interest groups=].
+    1. Let |key| be {|ig|'s [=interest group/trusted bidding signals coordinator=], |ig|'s
+      [=interest group/owner=]}.
+    1. If |keyValueInterestGroups| [=map/contain=] |key|, [=set/Append=] |ig| to |keyValueInterestGroups|
+      [|key|].
+    1. Otherwise:
+      1. Let |keyValueInterestGroups|[|key|] be an empty [=set=], whose [=map/values=] are [=strings=].
+      1. [=set/Append=] |ig| to |keyValueInterestGroups| [|key|].
+
+</div>
+
+<div algorithm>
+
+To <dfn>build trusted key value bidding signals request body</dfn> given a [=set=] |interestGroups|,
+an {{unsigned short}}-or-null |experimentGroupId|, an [=origin=] |topLevelOrigin|, a [=string=]
+|slotSizeQueryParam|, an [=origin=] |coordinator|, and an [=origin=] |owner|:
+
+1. Let |compressionGroups| be an empty [=list=], whose values are [=compression group=].
+1. Let |compressionGroupMap| be an empty [=map=], whose [=map/keys=] are integers and [=map/values=] are [=maps=].

```

Comment:
this part with several complex data structures is a bit hard to understand. Besides data type of its keys (and values), can you also mention what its keys (and possibly values) are supposed to be, like I think this map's key is compression group's id